_posts/2023-08-23-even-friendlier-sql.md

```diff
@@ -0,0 +1,596 @@
+
+Now compare that with the old way...
+
+```sql
+SELECT 
+     concat(
+          list_aggr(
+               string_split(
+                    UPPER('Make it stop'),
+               ' '),
+          'string_agg','.'),
+     '.') AS oof;
+```
+
+|      oof      |
+|:---|
+| MAKE.IT.STOP. |
+
+## Union by name
+
+DuckDB aims to blend the best of databases and dataframes. This new syntax is inspired by the [concat function in Pandas](https://pandas.pydata.org/docs/reference/api/pandas.concat.html). Rather than vertically stacking tables based on column position, columns are matched by name and stacked accordingly. Simply replace `UNION` with `UNION BY NAME` or `UNION ALL` with `UNION ALL BY NAME`. 
+
+For example, we had to add some new alien species proverbs in The Next Generation:
+```sql
+SELECT 
+     'Revenge is a dish best served cold' AS klingon_proverb 
+UNION ALL BY NAME 
+SELECT 
+     'You will be assimilated' AS borg_proverb,
+     'If winning is not important, why keep score?' AS klingon_proverb ;
+```
+
+|               klingon_proverb                |      borg_proverb       |
+|:---|:---|
+| Revenge is a dish best served cold           | NULL                    |
+| If winning is not important, why keep score? | You will be assimilated |
+
+This approach has additional benefits. As seen above, not only can tables with different column orders be combined, but so can tables with different numbers of columns entirely. This is helpful as schemas migrate, and is particularly useful for DuckDB’s [multi-file reading capabilities](https://duckdb.org/docs/data/multiple_files/combining_schemas.html#union-by-name). 
+
+## Insert by name
+
+Another common situation where column order is strict in SQL is when inserting data into a table. Either the columns must match the order exactly, or all of the column names must be repeated in two locations within the query. 
+
+Instead, add the keywords `BY NAME` after the table name when inserting. Any subset of the columns in the table in any order can be inserted.
+
+```sql
+INSERT INTO proverbs BY NAME 
+     SELECT 'Resistance is futile' AS borg_proverb;
+
+SELECT * FROM proverbs;
+```
+
+|               klingon_proverb                |      borg_proverb       |
+|:---|:---|
+| Revenge is a dish best served cold           | NULL                    |
+| If winning is not important, why keep score? | You will be assimilated |
+| NULL                                         | Resistance is futile    |
+
+## Dynamic PIVOT and UNPIVOT
+
+Historically, databases are not well suited for pivoting operations. However, DuckDB’s `PIVOT` and `UNPIVOT` clauses can create or stack dynamic column names for a truly flexible pivoting capability! In addition to that flexibility, DuckDB also provides both the SQL standard syntax and a friendlier shorthand. 
+
+For example, let’s take a look at some procurement forecast data just as the Earth-Romulan war was beginning:
+```sql
+CREATE OR REPLACE TABLE purchases AS 
```

Comment:
It would be 'cool' to have a easy way to express this examples (`FROM "{'some':'json'}"::JSON`), but I guess this is more a feature request than anything else.

Reply:
I went with your first format! Good thought!